File: 0075-sort-colors/0075-sort-colors.go
func sortColors(nums []int)  {
    n := len(nums)

    lo, hi, mid := 0, n-1, 0

    for mid <= hi {
        switch nums[mid]{
            case 0: 
                nums[lo], nums[mid] = nums[mid], nums[lo]
                lo++
                mid++
            case 1:
                mid++
            case 2:
                nums[hi], nums[mid] = nums[mid], nums[hi]
                hi--
        }
    }
}

File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int lo = 0, mid = 0 ,hi = n-1;

        while(mid <= hi){
            switch(nums[mid]){
                case 0:
                    swap(nums[lo], nums[mid]);
                    lo++;
                    mid++;
                    break;
                case 1:
                    mid++;
                    break;
                
                case 2:
                    swap(nums[mid], nums[hi]);
                    hi--;
                    break;
            }
        }
    }
};